blocks/initialization.py

```diff
@@ -147,21 +148,39 @@ def __init__(self, mult=1):
         self.mult = mult
 
     def generate(self, rng, shape):
-        if len(shape) != 2:
-            raise ValueError
-        rows, cols = shape
-        return self.mult * numpy.eye(rows, cols, dtype=theano.config.floatX)
+        if len(shape) == 1:
+            eye_matrix = numpy.eye(1, shape[0], dtype=theano.config.floatX)
```

Comment:
No idea whether it makes sense as an initialization, but it's the only correct extension to 1D cases. A `(n, 1)` or `(1, n)` truncated identity matrix would look the same.